src/main/java/com/dpgil/pathlinker/path_linker/internal/PathLinkerControlPanel.java

```diff
@@ -670,99 +670,116 @@ private void writeResult(ArrayList<Path> paths) {
 	}
 
 	/**
-	 * Creates a new network and view for the subgraph
+	 * Creates a new sub-network and sub-network view for the subgraph generated by the KSP
 	 */
-	private void createKSPSubgraphView() {
-
-		CyNetwork kspSubgraph = _model.getKspSubgraph();
+	private void createKSPSubgraphAndView() {
+		// creates task iterator and execute it to generate a sub-network from the original network
+	    // the bypass values and other styles from the original network will be pass down to the sub-network
+		TaskIterator subNetworkTask = _adapter.get_NewNetworkSelectedNodesAndEdgesTaskFactory()
+				.createTaskIterator(_model.getOriginalNetwork());
+		_adapter.getTaskManager().execute(subNetworkTask);
+
+		// This is a HACK: pausing execution with sleep to wait for the sub network to be created
+		try {
+			Thread.sleep(2000);
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
 
-		// creates the new network and its view
-		CyNetworkView kspSubgraphView = _networkViewFactory.createNetworkView(kspSubgraph);
-		_networkManager.addNetwork(kspSubgraph);
-		_networkViewManager.addNetworkView(kspSubgraphView);
+		// this is a hack as we can't access the new sub-network generated from the task
+		// loop through all the networks and find one with the highest SUID, 
+		// which is the sub-network we created
+		Set<CyNetwork> allNetworks = _networkManager.getNetworkSet();
+		long maxSUID = Integer.MIN_VALUE;
+		for(CyNetwork network : allNetworks) {
+			if (network.getSUID() > maxSUID) maxSUID = network.getSUID();
+		}
 
+		// Apply the new name to the sub-network
+		_kspSubgraph = _networkManager.getNetwork(maxSUID);
+		String subgraphName = "PathLinker-subnetwork-" + _model.getK() + "-paths";
```

Comment:
Can you change the "subnetwork" part of the name to "result-X" so the result number matches the number we set to the _result panel_ and _path index_ column?

Reply:
I'm going to make a new issue for this